src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+		} finally {
+			try {
+				restCommunicator.disconnect();
+			} catch (Exception e) {
+				if (this.logger.isErrorEnabled()) {
+					this.logger.error("error: Cannot disconnect from Rest communicator: " + this.host + " port: " + this.port);
+				}
+			}
+		}
+
+		extStats.setStatistics(stats);
+		extStats.setControllableProperties(advancedControllableProperties);
+
+		return Collections.singletonList(extStats);
+	}
+
+	/**
+	 * This method is used for populate all monitoring properties
+	 *
+	 * @param stats is the map that store all statistics
+	 */
+	private void populateMonitorCapabilities(Map<String, String> stats) {
+		stats.put(StatisticsProperty.DEVICE_INFORMATION.getName() + HASH + StatisticsProperty.DEVICE_MFG.getName(), deviceInfo.getDeviceMfg());
+		stats.put(StatisticsProperty.DEVICE_INFORMATION.getName() + HASH + StatisticsProperty.DEVICE_MODEL.getName(), deviceInfo.getDeviceModel());
+		stats.put(StatisticsProperty.DEVICE_INFORMATION.getName() + HASH + StatisticsProperty.DEVICE_SERIAL_NUMBER.getName(), deviceInfo.getDeviceSerialNumber());
+		stats.put(StatisticsProperty.DEVICE_INFORMATION.getName() + HASH + StatisticsProperty.DEVICE_FIRMWARE_VERSION.getName(), deviceInfo.getDeviceFirmwareVersion());
+		stats.put(StatisticsProperty.DEVICE_INFORMATION.getName() + HASH + StatisticsProperty.DEVICE_LAST_PRESET_RECALLED.getName(), this.getLastPresetRecalled());
+	}
+
+	/**
+	 * This method is used for populate all controlling properties
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 */
+	private void populateControlCapabilities(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		// Getting power status from device
+		PowerStatus powerStatus = getPowerStatus();
+		stats.put(Command.POWER.getName(), "");
+
+		assert powerStatus != null;
+		if (powerStatus.compareTo(PowerStatus.OFF) == 0) {
+			advancedControllableProperties.add(createSwitch(Command.POWER.getName(), 0, PowerStatus.OFF.getName(), PowerStatus.ON.getName()));
+		} else if (powerStatus.compareTo(PowerStatus.ON) == 0) {
+			advancedControllableProperties.add(createSwitch(Command.POWER.getName(), 1, PowerStatus.OFF.getName(), PowerStatus.ON.getName()));
+
+			// Zoom control
+			populateZoomControl(stats, advancedControllableProperties);
+
+			// Focus control
+			populateFocusControl(stats, advancedControllableProperties);
+
+			// AE control
+			populateAEControl(stats, advancedControllableProperties);
+
+			// WB control
+			populateWBControl(stats, advancedControllableProperties);
+
+			// Pan tilt control
+			populatePanTiltControl(stats, advancedControllableProperties);
+
+			// Preset control
+			populatePresetControl(stats, advancedControllableProperties);
+		}
+	}
+
+	//region Control device
+
+	/**
+	 * This method is used to control image process
```

Comment:
Please add more javadocs, stating what properties are (with examples maybe) and what this method does exactly. It would be quite tricky for other people to see what it does without looking at the code, and the goal of having javadocs is giving some information right off the bat, so it's easy to figure out whether this method specifically may be right place for troubleshooting, debugging etc. 
Other methods could use some more javadoc details also :)

Reply:
yes thank you, I will fix it